tests/Support/SupportCollectionTest.php

```diff
@@ -558,6 +558,20 @@ public function testMap()
         $this->assertEquals(['first' => 'first-rolyat', 'last' => 'last-llewto'], $data->all());
     }
 
+    public function testWalkWithModification()
+    {
+        $data = new Collection([['color' => 'red'],['color' => 'blue']]);
+        $data = $data->walk(function (&$item) { $item['color'] = strrev($item['color']); });
+        $this->assertEquals([['color' => 'der'],['color' => 'eulb']], $data->all());
+    }
+
+    public function testWalkWithoutModification()
+    {
+        $data = new Collection([['color' => 'red'],['color' => 'blue']]);
+        $data = $data->walk(function ($item) { $item['color'] = strrev($item['color']); });
+        $this->assertEquals([['color' => 'red'],['color' => 'blue']], $data->all());
```

Comment:
if you added a test to check the original collection is not modified, it would fail